### DISCUSSION TITLE
Using server-side pocketbase session in the client (Sveltekit)

### DISCUSSION THREAD

**Discussion by mikekok** [pocketbase/pocketbase#3837] [Q&A]
I'm using pocketbase for server-side authentication in a Sveltekit project.

So far so good, everything's working just fine when working with pocketbase in the server (`+layout.server.js` & `+page.server.js`).

I need to be able to also use the same pocketbase instance in the client though (`+page.js` & `+page.svelte`) in order to be able to load and update collections with [API rules](https://pocketbase.io/docs/api-rules-and-filters/) enabled.

What would be the best way to achieve this?

This is my server-side code that works as intended.
My purpose is to be able to run the code seen in the second code block (`+page.server.js`) in the client (`+page.js`), having the user already authenticated.

```javascript
// hooks.server.js

import { PUBLIC_DB_URL } from "$env/static/public"
import PocketBase from "pocketbase"
import { serializeNonPOJOs } from "$lib/utils/serializer"

export const handle = async ({ event, resolve }) => {
  event.locals.pb = new PocketBase(PUBLIC_DB_URL)
  event.locals.pb.authStore.loadFromCookie(
    event.request.headers.get("cookie") || ""
  )

  if (event.locals.pb.authStore.isValid) {
    event.locals.user = serializeNonPOJOs(event.locals.pb.authStore.model)
  } else {
    event.locals.user = undefined
  }

  const response = await resolve(event)

  response.headers.set(
    "set-cookie",
    event.locals.pb.authStore.exportToCookie({ secure: false })
  )

  return response
}
```

```javascript
// page.server.js

export const load = async ({ locals }) => {
  const getSales = async () => {
    const records = await locals.pb.collection("sales").getFullList({})
    return records
  }

  return {
    sales: getSales(),
  }
}
```

**Comment by ganigeorgiev** [comment]
In order to have "mixed" (aka. both server-side and client-side) regular user auth access, you'll have to export the cookie with `httpOnly: false` and from the client-side to load and set it using `document.cookie` (_or use framework cookie abstraction if there is such_):

```js
// client.pb.js
// Client PocketBase instance.
// When running in the browser is safe to be shared/global.

const pb = new PocketBase(PUBLIC_DB_URL)

pb.authStore.loadFromCookie(document.cookie)

pb.authStore.onChange((token, model) => {
    document.cookie = pb.authStore.exportToCookie({ httpOnly: false })
})

export default pb
```

The above for example can be placed in `hooks.client.js`.

---

With that said, why do you need the mixed access and why don't just use it only from the client-side?

A lot of developers seems to jump headfirst into JS SSR and I believe it is mistake. There is nothing wrong with JS SSR when you really need it and know what you are doing, but similar to this case, **when you start mixing regular users state between practically 2 different backends (the SvelteKit node server and PocketBase) things can get complicated very quickly** (_and this is not just about PocketBase and you'll stumble on the same thing with any 3rd party auth service that you decide to integrate with_) and in the end you may not need this complexity. I've tried to explain this in a little more details in https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7620032 and will try to find time in the near future to add sample guide(s) in the docs.

**Reply by ganigeorgiev** [reply]
_Side-note: If you use JS SDK v0.17.0+ you no longer need to call `serializeNonPOJOs` on the PocketBase requests result since the custom objects wrapping has been removed and we now return POJOs by default._
